packages/react-server/src/webpack-plugin/webpack-plugin.ts

```diff
@@ -34,6 +34,10 @@ export class ReactServerPlugin {
     const modules = this.modules(compiler);
     const virtualModules = new VirtualModulesPlugin(modules);
     (virtualModules as any).apply(compiler);
+
+    compiler.hooks.afterEnvironment.tap('ReactServerPlugin', () => {
+      compiler.watchFileSystem.paths.push(...Object.keys(modules));
+    });
```

Comment:
what if we make use of [`WatchIgnorePlugin`](https://github.com/webpack/webpack/blob/c181294865dca01b28e6e316636fef5f2aad4eb6/lib/WatchIgnorePlugin.js) and use it within this plugin on what to ignore.

Reply:
that could work